noxfile.py

```diff
@@ -0,0 +1,110 @@
+"""
+nox configuration for cookiecutter project template.
+"""
+
+import contextlib
+import os
+import tempfile
+from pathlib import Path
+
+import nox
+
+CI = os.environ.get('CI') is not None
+
+ROOT = Path('.')
+PYTHON_VERSIONS = ['3.9']
+PYTHON_DEFAULT_VERSION = PYTHON_VERSIONS[-1]
+
+nox.options.default_venv_backend = 'venv'
+nox.options.reuse_existing_virtualenvs = True
+
+
+# In CI, use Python interpreter provided by GitHub Actions
+if CI:
+    nox.options.force_venv_backend = 'none'
+
+
+CRUFTED_PROJECT_TMPDIR = None
+
+MD_PATHS = ['*.md']
+
+
+def run_readable(session, mode="fmt"):
+    session.run(
+        'docker',
+        'run',
+        '--rm',
+        '-v', f'{ROOT.absolute()}:/data',
+        '-w', '/data',
+        '-u', f'{os.geteuid()}:{os.getegid()}',
+        'ghcr.io/bobheadxi/readable:v0.5.0@sha256:423c133e7e9ca0ac20b0ab298bd5dbfa3df09b515b34cbfbbe8944310cc8d9c9',
+        mode, *MD_PATHS
+    )
+
+
+@nox.session(name='format', python=PYTHON_DEFAULT_VERSION)
+def format_(session):
+    """Lint the code and apply fixes in-place whenever possible."""
+    session.run('pip', 'install', '-e', '.[format]')
+    session.run('ruff', 'check', '--fix', '.')
+    run_readable(session, mode="fmt")
+
+
+@nox.session(python=PYTHON_DEFAULT_VERSION)
+def lint(session):
+    """Run linters in readonly mode."""
+    session.run('pip', 'install', '-e', '.[lint]')
+    session.run('ruff', 'check', '--diff', '.')
+    run_readable(session, mode="check")
+
+
+@contextlib.contextmanager
+def crufted_project(session):
+    session.run("pip", "install", "-e", ".")
+    global CRUFTED_PROJECT_TMPDIR
+    if not CRUFTED_PROJECT_TMPDIR:
+        CRUFTED_PROJECT_TMPDIR = tempfile.TemporaryDirectory(prefix="rt-crufted_")
+        session.log("Creating project in %s", CRUFTED_PROJECT_TMPDIR.name)
+        session.run("cruft", "create", ".", "--output-dir", CRUFTED_PROJECT_TMPDIR.name, "--no-input")
+        session.notify('cleanup_crufted_project')
```

Comment:
I'm not a huge fan of globals... wdyt about `@cache`?
![661](https://github.com/reef-technologies/cookiecutter-rt-django/assets/73276794/8f837a6c-4d16-4270-b0e2-4f181342ef33)

```python
from functools import cache

@cache
def cruft_project() -> Path:
    ...

@contextlib.contextmanager
def crufted_project(session):
    session.run("pip", "install", "-e", ".")
    crufted_project_path = cruft_project()
    project_path = Path(crufted_project_path.name) / "project"
    ...

@nox.session(python=PYTHON_DEFAULT_VERSION)
def cleanup_crufted_project(session):
    if not cruft_project.cache_info().currsize:
        return

    crufted_project_path = cruft_project()
    # workaround for docker-compose creating root-owned files
    rm_root_owned(session, crufted_project_path.name)
    crufted_project_path.cleanup()
    cruft_project.cache_clear()
```

Reply:
I though about using `@cache` here was that I need `session` param to create the project. 
I implemented another approach, which technically doesn't use globals, but it is somewhat hackish. I would use proper decorator class if not that it IMO made code less readable.